staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1368,22 +1369,37 @@ func (c *cacheWatcher) processEvents(ctx context.Context, initEvents []*watchCac
 	// consider increase size of result buffer in those cases.
 	const initProcessThreshold = 500 * time.Millisecond
 	startTime := time.Now()
-	for _, event := range initEvents {
+
+	initEventCount := 0
+	for {
+		event, err := cacheInterval.Next()
+		if err != nil {
+			// An error indicates that the cache
+			// interval has been invalidated and
+			// can no longer serve events.
+			klog.V(2).Info(err)
+			c.Stop()
```

Comment:
I suggest cleaning this up a bit and basically adding:
```
defer close(c.result)
defer c.Stop()
```

at the beginning of this function (after HandleCreash)
together with getting rid of it from process() function